tests/scripts/all.sh

```diff
@@ -3896,6 +3896,7 @@ component_test_aesni () { # ~ 60s
     make test programs/test/selftest CC=gcc CFLAGS='-Werror -Wall -Wextra -mpclmul -msse2 -maes'
     # check that we built intrinsics - this should be used by default when supported by the compiler
     ./programs/test/selftest aes | grep "AES note: using AESNI"
+    ./programs/test/selftest aes | grep -v "AES note: built-in implementation."
```

Comment:
Using `grep -v` like this will always succeed. You need to use `not grep "AES note: built-in implementation."` similar to the above negative checks.

Reply:
This test will also pass if using run-time detection, because under run-time detection, we will print `AES note: using AESNI` and we will not print `AES note: built-in implementation`. So it's not a good test of hardware-only.